ApplicationLibCode/ProjectDataModel/Flow/RimWellAllocationOverTimeCollection.cpp

```diff
@@ -49,9 +49,9 @@ RimWellAllocationOverTimeCollection::RimWellAllocationOverTimeCollection(
     for ( auto it = m_timeStepDates.rbegin(); it != m_timeStepDates.rend(); ++it )
     {
         const QDateTime& timeStep = *it;
-        if ( timeStepAndCalculatorPairs.contains( timeStep ) )
+        if ( auto tsPos = timeStepAndCalculatorPairs.find( timeStep ); tsPos != timeStepAndCalculatorPairs.end() )
         {
-            m_timeStepAndCalculatorPairs.emplace( timeStep, timeStepAndCalculatorPairs.at( timeStep ) );
+            m_timeStepAndCalculatorPairs.emplace( timeStep, tsPos->second );
             prevValidTimeStep = timeStep;
         }
```

Comment:
Good point regarding duplicated look up! Consider minor restructure for readability:

```
for ( auto rIt = m_timeStepDates.rbegin(); rIt != m_timeStepDates.rend(); ++rIt )
    {
        const QDateTime& timeStep             = *rIt;
        auto             timeStepCalculatorIt = timeStepAndCalculatorPairs.find( timeStep );
        if ( timeStepCalculatorIt != timeStepAndCalculatorPairs.end() )
        {
            m_timeStepAndCalculatorPairs.emplace( timeStep, timeStepCalculatorIt->second );
            prevValidTimeStep = timeStep;
        }
```

Reply:
> Consider minor restructure for readability

Thank you for your comments.  I have amended the PR per your advice.